src/apis/updateElement.js

```diff
@@ -15,9 +15,5 @@ WebViewer(...)
   });
  */
 export default store => (dataElement, overrides) => {
-  if (overrides !== null && typeof overrides !== 'object') {
```

Comment:
will this API be the only one where its second param is of an array?

Reply:
Yes it is. I updated the code to validate the input.

Reply:
doesn't it feel weird that this accepts 2 kinds of inputs?
An object with keys or an array?

Wouldn't it be a better idea to allow it to accept a parameter of one type?

Reply:
@achenPDFTron  I'm not sure if it's weird to users. I think function overloading is common to see for APIs. I guess the API signature should be whatever makes more sense to the user. If the second argument has to be an object, how would they use the API to update the palettes? Maybe something like
```javascript
instance.updateElement('colorPalette', {
  colors: ['#FFFFFF', '#...']
})
```

This can definitely work but I just feel like passing an array of colors is more intuitive...

Reply:
no i agree with you. passing an array of strings sounds more intuitive and easier in this case.

I think we should also consider how easy is it for us to maintain the API as well; not just whatever makes sense to the users.

My concern is how the second parameter is either an array or an object and how this could make it harder to debug and understand.

Ideally we should use an object or create a new function.

It looks like you have comments and documentations for now which could work.

I think in the far future, if we have more customize-able properties for the color palette, I think we should consider using an object.

Reply:
I agree what you said. Not sure if you've checked the latest code yet but I added some validations and break the main function body of `updateElement` into some small functions.

```javascript
switch (dataElement) {
  case 'colorPalette':
    overrides = validateColorPaletteOverrides(overrides);
    break;
  default:
    overrides = validateButtonOverrides(overrides);
}

if (overrides) {
  store.dispatch(actions.setCustomElementOverrides(dataElement, overrides));
}
```

I hope this can make it a bit clear about how this API is used? So to add more elements we just add more cases. For now I don't think it will be difficult to maintain because essentially this API just dispatch an action to the store and components just connect to that state to see if there're any custom overrides.

I agree that this API may get harder to harder to document once more elements can be customized with different way. So to make documentation easier and more clear I think we can split them into different APIs if necessary.

At least for me It's pretty hard to predict how things will change so I think if you are not strongly against the way the API(and the code structure) is written, we can let this PR go through.

Reply:
i think I am fine with it for now. i think if we have one more different dataElement type, we might want to rethink think.